src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,102 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int STOCK_ARRAY_LENGTH = 16;
+    private static final float MAX_FILL = 0.75f;
+    private static final int GROWTH_COEFICIENT = 2;
+    private Node<K,V>[] table;
+    private int size;
+
+    public MyHashMap() {
+        table = (Node<K, V>[]) new Node[STOCK_ARRAY_LENGTH];
+    }
 
     @Override
     public void put(K key, V value) {
-
+        Node<K, V> tempNode = getNode(key);
+        if (tempNode == null) {
+            growTableIfNeeded();
+            addNewNode(key, value);
+            size++;
+        } else {
+            tempNode.value = value;
+        }
     }
 
     @Override
     public V getValue(K key) {
-        return null;
+        Node<K,V> tempNode = getNode(key);
+        return tempNode != null ? tempNode.value : null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private int getBucket(K key) {
+        return Math.abs(hashOf(key) % table.length);
+    }
+
+    private Node<K,V> getNode(K key) {
+        Node<K,V> currentNode = table[getBucket(key)];
+        while (currentNode != null) {
+            if (currentNode.hash == hashOf(key) && keysAreEqual(currentNode.key, key)) {
```

Comment:
```suggestion
            if (keysAreEqual(currentNode.key, key)) {
```
Also instead of using your method - you can use Objects.equals(currentNode.key, key)